event/queue/eventQueue.go

```diff
@@ -0,0 +1,157 @@
+package queue
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+	"github.com/xmidt-org/glaukos/event/parsing"
+	"github.com/xmidt-org/webpa-common/basculechecks"
+	"github.com/xmidt-org/webpa-common/logging"
+	"github.com/xmidt-org/webpa-common/semaphore"
+	"github.com/xmidt-org/wrp-go/v3"
+	"go.uber.org/fx"
+)
+
+var (
+	defaultLogger = log.NewNopLogger()
+)
+
+const (
+	defaultMinMaxWorkers = 5
+	defaultMinQueueSize  = 5
+)
+
+// QueueConfig configures the glaukos queue used to parse incoming events from Caduceus
+type Config struct {
+	QueueSize  int
+	MaxWorkers int
+}
+
+type EventQueue struct {
+	queue   chan wrp.Message
+	workers semaphore.Interface
+	wg      sync.WaitGroup
+	logger  log.Logger
+	config  Config
+	parsers parsing.ParsersIn
+	metrics QueueMetricsIn
+}
+
+func newEventQueue(config Config, parsers parsing.ParsersIn, metrics QueueMetricsIn, logger log.Logger) (*EventQueue, error) {
```

Comment:
Usually new functions are meant to be like normal Go code and compatible outside of `uber fx`, so I'm a little concerned about the `_In` structs included here.  Usually our new functions just require `Measures` right?